sinks/interamap/src/real_time_parser.py

```diff
@@ -0,0 +1,136 @@
+import queue
+import sys
+from omnibus import Receiver
+from PySide6.QtCore import QThread, Signal
+
+try:
+    from src.data_struct import Point_GPS, Info_GPS
+except ImportError:
+    from data_struct import Point_GPS, Info_GPS
+
+def parse_gps_data(gps, data):
+    timestamp = "{hrs:02}:{mins:02}:{secs:02}.{dsecs:02}".format(                                
+        **gps.get("GPS_TIMESTAMP", {"hrs": 0, "mins": 0, "secs": 0, "dsecs": 0}))
+    latitude = gps.get("GPS_LATITUDE", {"degs": 0, "mins": 0, "dmins": 0})
+    longitude = gps.get("GPS_LONGITUDE", {"degs": 0, "mins": 0, "dmins": 0})
+    altitude = gps.get("GPS_ALTITUDE", {"altitude": 0, "daltitude": 0})
+    num_sats = gps.get("GPS_INFO", {"num_sats": 0})["num_sats"]
+    boardId = data.get("board_id")
+
+    # Convert latitude and longitude to decimal degrees
+    lat = convert_to_decimal_degrees(latitude)
+    lon = convert_to_decimal_degrees(longitude)
+
+    if boardId == "PROCESSOR":
+        # If the board ID is PROCESSOR, the timestamp is not available
+        timestamp = None
+        lon, lat = switch_numbers_keep_sign(lon, lat) # NOTE: This is for fix a mistake in the PROCESSOR data
+        
+    # Combine altitude and decimal altitude
+    he = altitude["altitude"] + altitude["daltitude"] / 100
+
+    point = Point_GPS(lon=lon, lat=lat, he=he, num_sats=num_sats, time_stamp=timestamp, board_id=boardId)
+
+    return point
+
+def parse_gps_info(data):
+    info = data.get("data",{"num_sats": 0, "quality": 0})
+    num_sats = info["num_sats"]
+    quality = info["quality"]
+    boardId = data.get("board_id")
+
+    info = Info_GPS(num_sats=num_sats, quality=quality, board_id=boardId)
+
+    return info
+
+def switch_numbers_keep_sign(a, b):
+    # Swap absolute values without changing signs
+    new_a = abs(b) if a >= 0 else -abs(b)
+    new_b = abs(a) if b >= 0 else -abs(a)
+    return new_a, new_b
+
+def convert_to_decimal_degrees(coord):
+    degs = coord["degs"]
+    mins = coord["mins"] / 60
+    dmins = coord["dmins"] / 3600
```

Comment:
We send dmin by hundredths of a minute, so it should be divided by 6000 not 3600